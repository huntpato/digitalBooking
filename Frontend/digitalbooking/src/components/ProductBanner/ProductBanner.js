import React from 'react'
import { useHistory } from "react-router-dom" 
import styles from "./ProductBanner.module.css"

export const ProductBanner = (props) => {
    let history = useHistory();
    const {title, subtitle}= props;
    const { header, headerContainer, titleContainer } = styles;
    return (
        <section className={header}>
        <div className={headerContainer}>
          <div className={titleContainer}>
            <h5>{ title }</h5>
            <h3>{ subtitle }</h3>
          </div>
          <i className="fas fa-arrow-left" onClick={() => history.goBack()}></i>
        </div>
      </section>
    )
}

