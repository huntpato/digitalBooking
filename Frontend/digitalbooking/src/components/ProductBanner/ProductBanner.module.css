.header {
    background-color: var(--color-typo2);
    color: var(--background-white);
    padding-top: 30px;
    display:flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  
  .headerContainer{
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 15px;
    width:90vw;
  }

  .titleContainer{
    width: 90%;
  }
  
  .header h5 {
    padding-bottom: 4px;
    width:90%
  }

  .header h3{
    width:100%;
  }
  
  .header i {
    color: white; 
  }