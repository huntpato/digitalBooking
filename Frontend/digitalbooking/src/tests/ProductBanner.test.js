import React from 'react';
import { shallow } from 'enzyme';
import { ProductBanner } from '../components/ProductBanner/ProductBanner'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <ProductBanner />', ()=> {

    const wrapper= shallow(<ProductBanner title="Hotel" subtitle="subtitulo" />);

    test('Comprobar que <ProductBanner /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Comprobar el titulo correctamente.', () => {
    
        const title = wrapper.find('h5').text();
        expect(title).toContain("Hotel");
      
    })

    test('Comprobar el subtitulo correctamente.', () => {
    
        const title = wrapper.find('h3').text();
        expect(title).toContain("subtitulo");
      
    })

    

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div').at(0);
        expect(div.hasClass('headerContainer')).toBe(true);
    })

})
