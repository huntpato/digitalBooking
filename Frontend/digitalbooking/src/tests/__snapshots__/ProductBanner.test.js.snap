// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <ProductBanner /> Comprobar que <ProductBanner /> renderiza correctamente. 1`] = `
<section
  className="header"
>
  <div
    className="headerContainer"
  >
    <div
      className="titleContainer"
    >
      <h5>
        Hotel
      </h5>
      <h3>
        subtitulo
      </h3>
    </div>
    <i
      className="fas fa-arrow-left"
      onClick={[Function]}
    />
  </div>
</section>
`;
